#import faulthandler; faulthandler.enable()
import pysgpp
import math
import sys
import numpy as np
import matplotlib as plt

def circle(radius, dim):
    